saleor/order/actions.py

```diff
@@ -576,15 +582,17 @@ def _increase_order_line_quantity(order_lines_info):
     OrderLine.objects.bulk_update(order_lines, ["quantity_fulfilled"])
 
 
-@traced_atomic_transaction()
 def fulfill_order_lines(
     order_lines_info: Iterable["OrderLineInfo"],
     manager: "PluginsManager",
     allow_stock_to_be_exceeded: bool = False,
 ):
     """Fulfill order line with given quantity."""
-    _decrease_stocks(order_lines_info, manager, allow_stock_to_be_exceeded)
-    _increase_order_line_quantity(order_lines_info)
+    # transaction ensures that there is a consistency between quantities in order line
+    # and stocks
+    with traced_atomic_transaction():
+        _decrease_stocks(order_lines_info, manager, allow_stock_to_be_exceeded)
+        _increase_order_line_quantity(order_lines_info)
 
 
 @traced_atomic_transaction()
```

Comment:
Is it expected? You added new transaction below

Reply:
thanks, good catch. It seems I've missed it somehow